source/slides/08_systemd.rst

```diff
@@ -0,0 +1,431 @@
+``oneshot``
+  This type is similar to simple, but the process exits before starting
+  consequent units.
+``notify``
+  This type is similar to simple, but consequent units are started only after a
+  notification message.
+
+``[Service]`` Section Example: Postfix
+--------------------------------------
+
+::
+
+  [Service]
+  Type=forking
+  PIDFile=/var/spool/postfix/pid/master.pid
+  EnvironmentFile=-/etc/sysconfig/network
+  ExecStartPre=-/usr/libexec/postfix/aliasesdb
+  ExecStartPre=-/usr/libexec/postfix/chroot-update
+  ExecStart=/usr/sbin/postfix start
+  ExecReload=/usr/sbin/postfix reload
+  ExecStop=/usr/sbin/postfix stop
+
+``[Install]`` Section Options
+-----------------------------
+
+.. csv-table::
+  :header: Option, Description
+  :widths: 5, 10
+
+  ``Alias``, Provides a space-separated list of additional names for the unit.
+  ``RequiredBy``, A list of units that depend on the unit.
+  ``WantedBy``, "A list of units that weakly depend on the unit. When this unit
+  is enabled, the units listed in ``WantedBy`` gain a ``Want`` dependency on the
+  unit."
+  ``Also``, "Specifies a list of units to be installed or uninstalled along with
+  the unit."
+
+``[Install]`` Section Example: Postfix
+--------------------------------------
+
+::
+
+  [Install]
+  WantedBy=multi-user.target
+
+Creating your own Units
+-----------------------
+
+``/etc/systemd/system/hello-world.service``
+
+::
+
+  [Unit]
+  Description=Hello World
+
+  [Service]
+  ExecStart=/bin/sh -c 'echo Hello World; sleep 5'
+  StandardOutput=syslog
+  StandardError=syslog
+
+  [Install]
+  WantedBy=multi-user.target
+
+Instantiated (Parameterized) Units
+----------------------------------
+
+* Possible to to instantiate multiple units from a single template config
+* The ``@`` character is used to mark the template
+
+Named the following way::
+
+  # Unit service name
+  template_name@instance_name.service
+
+  # Template file name
+  template_name@.service
+
+``template_name``
+  Name of the template config file
+``instance_name``
+  Name of the unit instance
+
+Example: Login Prompts (getty)
+
+Instantiated Units: Specifiers
+------------------------------
+
+.. csv-table::
+  :header: Unit Specifier, Description
+  :widths: 5, 10
+
+  ``%n``, Full unit name (i.e. getty@ttyS1.service)
+  ``%p``, Prefix name (i.e. ``template_name``)
+  ``%i``, Instance name (i.e. ``instance_name``)
+  ``%H``, Host name
+
+::
+
+  [Unit]
+  Description=Getty on %I
+  ...
+  [Service]
+  ExecStart=-/sbin/agetty --noclear %I $TERM
+  ...
+
+Enabling and using instantiated units
+-------------------------------------
+
+Enable via a symlink:
+
+.. code-block:: bash
+
+  ln -s /usr/lib/systemd/system/getty@.service \
+    /etc/systemd/system/getty@tty1.service
+
+  # Add to the default.target
+  ln -s /usr/lib/systemd/system/getty@.service \
+    /etc/systemd/system/default.target.wants/getty@tty1.service
+
+Anything in the ``.wants`` directory automatically added as dependencies for
```

Comment:
You might want to clarify that it's merged with the existing `wants` list. A great way to do this is to show `multi-user.target`

```
jme@pegafish:pubstack:>grep Wants /lib/systemd/graphical.target
Wants=display-manager.service
jme@pegafish:pubstack:>systemctl show graphical.target | grep Wants
Wants=cpufrequtils.service speech-dispatcher.service systemd-update-utmp-runlevel.service lightdm.service postfix.service cgroupfs-mount.service loadcpufreq.service accounts-daemon.service hddtemp.service virtualbox.service irqbalance.service sysstat.service gdomap.service minissdpd.service
```